src/components/elements/ButtonLink.tsx

```diff
@@ -1,30 +1,36 @@
+import { SvgIconComponent } from '@mui/icons-material';
 import { Button, ButtonProps } from '@mui/material';
-import { forwardRef } from 'react';
+import { forwardRef, Ref } from 'react';
 import { Link, LinkProps } from 'react-router-dom';
 
-export type ButtonLinkProps = Omit<ButtonProps, 'href'> &
+export type ButtonLinkProps = Omit<ButtonProps, 'href' | `on${string}`> &
```

Comment:
Unioning `ButtonProps` with `LinkProps` creates a bunch of prop type issues related to the underlying ref types, `HTMLAnchorElement` and `HTMLButtonElement`. The upstream callers of `<ButtonLink>` didn't need to pass any of these `on*` handlers anyway (onCopy, onPaste etc.) so I just omitted them all by way of this handy masking functionality, which I learned about [here](https://stackoverflow.com/questions/53501721/typescript-exclude-property-key-when-starts-with-target-string).

<details>

```

src/components/layout/PrintViewButton.tsx:46:8 - error TS2769: No overload matches this call.
  Overload 1 of 3, '(props: { href: string; } & ButtonOwnProps & Omit<ButtonBaseOwnProps, "classes"> & CommonProps & Omit<Omit<DetailedHTMLProps<AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement>, "ref"> & { ...; }, "color" | ... 24 more ... | "startIcon">): Element', gave the following error.
    Property 'href' is missing in type '{ children: string | number | boolean | ReactElement<any, string | JSXElementConstructor<any>> | Iterable<ReactNode> | null; ... 305 more ...; leftAlign?: boolean | undefined; }' but required in type '{ href: string; }'.
  Overload 2 of 3, '(props: { component: ElementType<any, keyof IntrinsicElements>; } & ButtonOwnProps & Omit<ButtonBaseOwnProps, "classes"> & CommonProps & Omit<...>): Element | null', gave the following error.
    Type '{ children: string | number | boolean | ReactElement<any, string | JSXElementConstructor<any>> | Iterable<ReactNode> | null; ... 305 more ...; leftAlign?: boolean | undefined; }' is not assignable to type '{ component: ElementType<any, keyof IntrinsicElements>; }'.
      Types of property 'component' are incompatible.
        Type 'ElementType<any, keyof IntrinsicElements> | undefined' is not assignable to type 'ElementType<any, keyof IntrinsicElements>'.
          Type 'undefined' is not assignable to type 'ElementType<any, keyof IntrinsicElements>'.
  Overload 3 of 3, '(props: DefaultComponentProps<ExtendButtonBaseTypeMap<ButtonTypeMap<{}, "button">>>): Element | null', gave the following error.
    Type 'MouseEventHandler<HTMLAnchorElement>' is not assignable to type 'MouseEventHandler<HTMLButtonElement>'.
      Type 'HTMLAnchorElement' is missing the following properties from type 'HTMLButtonElement': disabled, form, formAction, formEnctype, and 13 more.

46       <Button
          ~~~~~~

  node_modules/@mui/material/ButtonBase/ButtonBase.d.ts:107:12
    107   props: { href: string } & OverrideProps<ExtendButtonBaseTypeMap<TypeMap>, 'a'>,
                   ~~~~
    'href' is declared here.
  node_modules/@types/react/index.d.ts:2489:9
    2489         onClick?: MouseEventHandler<T> | undefined;
                 ~~~~~~~
    The expected type comes from property 'onClick' which is declared here on type 'IntrinsicAttributes & ButtonOwnProps & Omit<ButtonBaseOwnProps, "classes"> & CommonProps & Omit<...>'

```
</details>